cms/static/js/views/settings/main.js

```diff
@@ -236,6 +236,18 @@ var DetailsView = ValidatingView.extend({
             this.model.set('instructor_info', {instructors: instructors});
             this.showNotificationBar();
             break;
+        case 'course-instructor-image-' + index:
+            instructors = this.model.get('instructor_info').instructors.slice(0);
+            instructors[index].image = $(event.currentTarget).val();
+            this.model.set('instructor_info', {instructors: instructors});
+            this.showNotificationBar();
+
+            // Wait to set the image src until the user stops typing
+            clearTimeout(this.imageTimer);
+            this.imageTimer = setTimeout(function() {
+                $(event.currentTarget).parents().eq(2).find('img').attr('src', $(event.currentTarget).val());
+            }, 1000);
+            break;
```

Comment:
Ah, I see that this is a bit of special case because of the instructors list. I'm going to at least break out the bit that updates the image preview into a separate function and use that here.